crates/client/db/src/bonsai_db.rs

```diff
@@ -22,23 +28,29 @@ impl DatabaseKeyMapping {
     }
 }
 
-pub struct BonsaiDb<'db> {
-    db: &'db DB,
+pub struct BonsaiDb {
+    db: Arc<DB>,
     /// Mapping from `DatabaseKey` => rocksdb column name
     column_mapping: DatabaseKeyMapping,
-    // snapshots: BTreeMap<BasicId, SnapshotWithThreadMode<'db, DB>>,
+    snapshots: Arc<Snapshots>,
     write_opt: WriteOptions,
 }
 
-impl<'db> BonsaiDb<'db> {
-    pub(crate) fn new(db: &'db DB, column_mapping: DatabaseKeyMapping) -> Self {
+impl BonsaiDb {
+    pub(crate) fn new(db: Arc<DB>, snapshots: Arc<Snapshots>, column_mapping: DatabaseKeyMapping) -> Self {
         let mut write_opt = WriteOptions::default();
         write_opt.disable_wal(true);
-        Self { db, column_mapping, write_opt }
+        Self { db, column_mapping, write_opt, snapshots }
     }
 }
 
-impl BonsaiDatabase for BonsaiDb<'_> {
+impl fmt::Debug for BonsaiDb {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "<database>")
```

Comment:
So, I don't use this fmt::Debug. This is used on the bonsai-trie side for debug printing the whole database, but in a real world where we have more than a few entries in db, i don't want to dump the whole database to the terminal by accident

Reply:
the trait looks like
trait BonsaiDb: fmt::Debug
which forces us to implement Debug, the only way to avoid forcin people to impl Debug here would be a separate copy-pasted definition of the whole trait with #[cfg(not(test)))] and without the fmt::Debug bound, which sounds silly

Reply:
> i don't want to dump the whole database to the terminal by accident

Yes, of course. My point was simply have "BonsaiDb" in the debug output and not a generic "database" so that any it's clear to anyone looking at the debug output what the type is.